srcpkgs/julia/patches/julia-hardcoded-libs.patch

```diff
@@ -0,0 +1,43 @@
+diff --git a/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl b/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl
+index a7ca666b23..b84a42f4e7 100644
+--- a/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl
++++ b/stdlib/MbedTLS_jll/src/MbedTLS_jll.jl
+@@ -31,9 +31,9 @@ elseif Sys.isapple()
+     const libmbedtls = "@rpath/libmbedtls.13.dylib"
+     const libmbedx509 = "@rpath/libmbedx509.1.dylib"
+ else
+-    const libmbedcrypto = "libmbedcrypto.so.5"
+-    const libmbedtls = "libmbedtls.so.13"
+-    const libmbedx509 = "libmbedx509.so.1"
++    const libmbedcrypto = "libmbedcrypto.so"
++    const libmbedtls = "libmbedtls.so"
```

Comment:
This only works on Arch because they don't have devel packages. The unversioned sonames are all in our -devel packages, not the normal ones - this solution isn't portable across distros.

For that matter, if they're using dlopen, they *should* very much care about the soname version: ABI mismatches cause all sorts of bugs. If my understanding of this patch is correct, I am against it. Will look at the upstream bugs to see if I can add anything there.

Reply:
It works for us as well; I've confirmed things work locally without mbedtls-devel installed. Julia appears to use these variables to generate the symbolic links in `/usr/lib/julia`.

```sh
❮ ls -l /usr/lib/julia/libmbedcrypto.so 
lrwxrwxrwx 1 root root 29 Apr  6 19:03 /usr/lib/julia/libmbedcrypto.so -> /usr/lib64/libmbedcrypto.so.3
```

Reply:
I agree that forcing `-devel` dependencies just to drop versions from shared libraries is not desirable.

As for `dlopen`, the question is how Julia handles these libraries after they are loaded. If the runtime handles the possibility of missing symbols gracefully, maybe this isn't such a big deal. If not, this could be a huge deal.